### ISSUE TITLE
Feature Request: Transactions without containers

### ISSUE THREAD

**Issue by lsimoneau** [dry-rb/dry-transaction#53]
My team is keen to adopt dry-transactions in some of our systems, but some developers are a bit more hesitant about introducing containers. Currently you can't have one without the other, so even if you're not using a system-wide container for everything, you still need one for your transactions. Without adopting the whole dry-rb approach, the container ends up being an extra layer of indirection between the transactions and the operations that doesn't deliver much benefit, which we'd prefer to avoid.

Currently we can work around this using inline steps, either like:

```ruby
MyTransaction = Dry::Transaction(container: []) do
  step :do_a_thing, with: -> (input) { MyOperation.call(input) }
end
```

or

```ruby
MyTransaction = Dry::Transaction(container: []) do
  step :do_a_thing, with: MyOperation.method(:call)
end
```

I'd have expected objects responding to `#call` to be usable directly in steps without wrapping them in a lambda, but currently that's not the case.

In any case, we'd really like to be able to use the transaction DSL without containers, as in the above examples the `container:` key is only there to satisfy the API, and is just cruft.

**Comment by timriley** [comment]
@lsimoneau Thanks for your thoughts here (and for giving dry-transaction a try!)

I'm up for doing this, since like you describe, it should make adoption easier for people with different kinds of application setups.

Your request comes at a good time, since I actually started thinking about similar ease-of-use improvements over in #52, just this morning. If it's OK with you, I'd probably incorporate this feature into the work on that (or do it immediately after).

Just an early thought while I'm here: it might actually be good if we made this a first-class, configurable things via step _resolvers_. This might open up some interesting uses down the track.

**Comment by timriley** [comment]
Hi @lsimoneau – we've just resolved this as part of #58 and I hope to make a release tomorrow.

You can see over in the soon-to-be-released doc updates how you can work with container-less transactions: https://github.com/dry-rb/dry-rb.org/blob/e9c56234edfb8de9beede674b045e1c584df5f5d/source/gems/dry-transaction/basic-usage.html.md#defining-a-transaction-with-local-operations

Hope this satisfies your requirements! I'll close this issue now but feel free to reopen this one or another one if you see any deficiencies or bugs :)

**Comment by EugZol** [comment]
What about something like this?

```
class MyStep
  include Dry::Transaction::Operation

  def call(...)
    # ...
  end
end

class MyTransaction
  include Dry::Transaction

  step :do_something, with: MyStep
end
```

Is it possible?
